packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -36,12 +37,30 @@ const propTypes = {
 // eslint-disable-next-line react/prefer-stateless-function
 class DatePickerInput extends React.Component {
   render() {
+    // If a name attribute is provided in inputAttributes, it will be used as the name attribute in the hidden input.
+    // If no name attribute is provided, a default name of 'terra-date' will be used.
+    let nameAttr = 'terra-date';
```

Comment:
As discussed in post scrum, a required name prop is added and we expect that the consumer will pass in a unique name for each instance of the input.